streams/src/main/java/org/apache/kafka/streams/processor/internals/TopologyMetadata.java

```diff
@@ -83,6 +100,7 @@ public TopologyMetadata(final InternalTopologyBuilder builder,
         } else {
             builders.put(UNNAMED_TOPOLOGY, builder);
         }
+        getStreamThreadCount = () -> getNumStreamThreads(config);
```

Comment:
Why we need to initialize this supplier in the constructor (ditto below)? This variable is only called in `reachedVersion` when stream threads have been initialized already.

Reply:
This is because the number of threads can change. If they do we don't want to be stuck if a thread is removed

Reply:
removed